plugins/router/f5/f5.go

```diff
@@ -0,0 +1,1743 @@
+		HttpHost bool `json:"httpHost,omitempty"`
+
+		// HttpUri indicates that the condition must match on the request URI.
+		HttpUri bool `json:"httpUri,omitempty"`
+
+		// PathSegment, used with HttpUri, indicates that the condition must match
+		// on a particular path segment from the request URI.
+		PathSegment bool `json:"pathSegment,omitempty"`
+
+		// Index indicates which component of an item (e.g., which segment of
+		// a path) this condition checks.
+		Index int `json:"index"`
+
+		// Equals indicates that the condition tests for equality.
+		Equals bool `json:"equals"`
+
+		// Request indicates that the rule matches on requests as opposed to
+		// responses.
+		Request bool `json:"request"`
+
+		// Host, used with HttpHost, indicates that the rule should match against
+		// the vhost (as opposed to the port or both the vhost and the port).
+		Host bool `json:"host,omitempty"`
+
+		// Values specifies items for matching such as pathnames or hostnames.
+		Values []string `json:"values"`
+	}{
+		Name:            "0",
+		CaseInsensitive: true,
+		HttpHost:        true,
+		Host:            true,
+		Index:           0,
+		Equals:          true,
+		Request:         true,
+		Values:          []string{hostname},
+	}
+
+	err = f5.post(conditionsUrl, conditionsPayload, nil)
+	if err != nil {
+		return err
+	}
+
+	if pathname != "" {
+		// Each segment of the pathname must be added to the rule as a separate
+		// condition.
+		segments := strings.Split(pathname, "/")
+		conditionsPayload.HttpHost = false
+		conditionsPayload.Host = false
+		conditionsPayload.HttpUri = true
+		conditionsPayload.PathSegment = true
+		for i, segment := range segments[1:] {
+			idx := fmt.Sprintf("%d", i+1)
+			conditionsPayload.Name = idx
+			conditionsPayload.Index = i + 1
+			conditionsPayload.Values = []string{segment}
+			err = f5.post(conditionsUrl, conditionsPayload, nil)
+			if err != nil {
+				return err
+			}
+		}
+	}
+
+	actionsUrl := fmt.Sprintf("https://%s/mgmt/tm/ltm/policy/%s/rules/%s/actions",
+		f5.host, policyname, escapedRulename)
+
+	actionsPayload := struct {
+		// Name is the name of the action.  We generally just use "0".
+		Name string `json:"name"`
+
+		// Forward indicates that the connection should be forwarded.
+		Forward bool `json:"forward"`
+
+		// Pool, used with Forward and Select, indicates a pool to which the
+		// connection should be forwarded.
+		Pool string `json:"pool"`
+
+		// Request indicates that the action takes effect on requests as opposed to
+		// responses.
+		Request bool `json:"request"`
+
+		// Select indicates that the action selects a destination for the connection
+		// (as opposed to resetting it).
+		Select bool `json:"select"`
+
+		// Vlan is the vlan associated with the destination.
+		Vlan int `json:"vlanId"`
+	}{
+		Name:    "0",
+		Forward: true,
+		Pool:    fmt.Sprintf("/Common/%s", poolname),
+		Request: true,
+		Select:  true,
+		Vlan:    0,
+	}
```

Comment:
Similar comment to above - having the struct definition in a separate file improves readability in this function.